tests/e2e/test_app.py

```diff
@@ -104,15 +104,15 @@ def test_health(client):
     assert health_response.is_success
     assert (
             health_response.json()
-            == HealthStatus(pinecone_status="OK", llm_status="OK").dict()
+            == HealthStatus(pinecone_status="OK", llm_status="OK").model_dump()
```

Comment:
Again, I think that 
```suggestion
            == dict(HealthStatus(pinecone_status="OK", llm_status="OK"))
```
is more readable.  
But totally your call

Reply:
As I said model_dump is recursive, so better to stick onto that.